src/assets/src/hooks/useGoogleAnalytics.ts

```diff
@@ -1,20 +1,22 @@
-import { useState, useEffect } from 'react'
-import GoogleAnalytics from 'react-ga'
+import { useState, useEffect } from 'react';
+import GoogleAnalytics from 'react-ga4';
+import { useLocation } from 'react-router-dom';
 
 export const useGoogleAnalytics = (googleAnalyticsId?: string, debug?: boolean) => {
+    let location = useLocation();
     const [initialized, setInitialized] = useState(false);
     const [previousPage, setPreviousPage] = useState(null as string | null);
 
     if (googleAnalyticsId && !initialized) {
         setInitialized(true);
-        GoogleAnalytics.initialize(googleAnalyticsId, { debug });
+        GoogleAnalytics.initialize(googleAnalyticsId, { testMode: debug });
```

Comment:
the `debug` is currently tied up so Django debug setting, which means when Django.debug = true then GA tracking won't happen, This is doesn't effect Prod. But when testing Non-prod the debug option will be turned on in that case this could cause for confusion if particularly testing for GA events. This should be documented along with GA configuration options

Reply:
It actually looks like we use `DEBUG` with `False` in all our deployments, so it's probably simplest to test in one of our non-prod environments. I noted this in the addition to the `README` I made: https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/412/commits/fbbc22f94db3d9302367a91f11be15afed1bfa5e

It's not a bad idea to have some other kind of flag like `GA_DEBUG` so we can more effectively do local testing, but I think this isn't something we're going to have worry about often.